src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityWorldAccelerator.java

```diff
@@ -34,26 +38,49 @@
 
 import javax.annotation.Nullable;
 import java.util.List;
+import java.util.Map;
 import java.util.function.Supplier;
 
 import static gregtech.api.capability.GregtechDataCodes.IS_WORKING;
 import static gregtech.api.capability.GregtechDataCodes.SYNC_TILE_MODE;
 
 public class MetaTileEntityWorldAccelerator extends TieredMetaTileEntity implements IControllable {
 
-    private static Class<?> cofhTileClass;
+    private static final String COFH_CLASS_NAME = "cofh.thermalexpansion.block.device.TileDeviceBase";
+
+    private static final Map<String, Class<?>> blacklistedClasses = new Object2ObjectOpenHashMap<>();
+    private static final Map<Class<? extends TileEntity>, Boolean> blacklistCache = new Object2BooleanOpenHashMap<>();
+    private static boolean gatheredClasses = false;
 
     private static boolean considerTile(TileEntity tile) {
-        // TODO interface for this?
-        if (tile instanceof IGregTechTileEntity || tile instanceof TileEntityPipeBase) {
-            return false;
-        }
-        if (cofhTileClass == null) {
+        if (tile instanceof IGregTechTileEntity || tile instanceof TileEntityPipeBase) return false;
+
+        if (!gatheredClasses) {
+            for (String name : ConfigHolder.machines.worldAcceleratorBlacklist) {
+                if (!blacklistedClasses.containsKey(name)) {
+                    try {
+                        blacklistedClasses.put(name, Class.forName(name));
+                    } catch (ClassNotFoundException ignored) {
+                        GTLog.logger.warn("Could not find class {} for World Accelerator Blacklist.", name);
+                    }
+                }
+            }
             try {
-                cofhTileClass = Class.forName("cofh.thermalexpansion.block.device.TileDeviceBase");
-            } catch (Exception ignored) {}
+                blacklistedClasses.put(COFH_CLASS_NAME, Class.forName(COFH_CLASS_NAME));
+            } catch (ClassNotFoundException ignored) {/**/}
+
+            gatheredClasses = true;
         }
-        return cofhTileClass == null || !cofhTileClass.isInstance(tile);
+
+        return blacklistCache.computeIfAbsent(tile.getClass(), c -> {
```

Comment:
Don't we want to check before this `computeIfAbsent` call if the tile entity is already present in the blacklistCache? This blacklist attempt will fail if it scans a TE that is already included in the blacklistCache, because it is using a `computeIfAbsent`. There should be a simple, 

```java
if (blacklistCache.hasKey(tile.class ) {
    return false;
}
```

check before this `computeIfAbsent` call.